pkg/kubectl/history.go

```diff
@@ -131,6 +200,29 @@ func (h *DeploymentHistoryViewer) ViewHistory(namespace, name string, revision i
 	})
 }
 
+// controlledHistories returns all ControllerRevisions controlled by the given DaemonSet
+func controlledHistories(c clientset.Interface, ds *extensions.DaemonSet) ([]*appsv1beta1.ControllerRevision, error) {
+	var result []*appsv1beta1.ControllerRevision
+	selector, err := metav1.LabelSelectorAsSelector(ds.Spec.Selector)
+	if err != nil {
+		return nil, err
+	}
+	versionedClient := versionedClientsetForDaemonSet(c)
+	historyList, err := versionedClient.Apps().ControllerRevisions(ds.Namespace).List(metav1.ListOptions{LabelSelector: selector.String()})
+	if err != nil {
+		return nil, err
+	}
+	for i := range historyList.Items {
```

Comment:
Because go reuses the same memory location for each iteration, and I'm storing `&history` in `result`.
See [the first go landmine](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)